web/packages/shared/components/Notification/Notification.tsx

```diff
@@ -113,79 +114,145 @@ export function Notification(props: NotificationProps) {
       onClick={toggleIsExpanded}
       {...styleProps}
     >
-      <Flex alignItems="center" mr={1} minWidth="0" width="100%">
-        <Icon color={getColor(theme)} mr={3} size="medium" />
-        {getRenderedContent(item.content, isExpanded, removeIcon)}
+      <Flex gap={2} flexDirection="column">
+        <Flex flexDirection="row" gap={2}>
+          <NotificationIcon
```

Comment:
Would it be possible to make it so that the icon always aligns to the top?

<details>
<summary>Patch</summary>

```diff
diff --git a/web/packages/design/src/Icon/Icon.tsx b/web/packages/design/src/Icon/Icon.tsx
index fea9c16ef2..52d6acbf86 100644
--- a/web/packages/design/src/Icon/Icon.tsx
+++ b/web/packages/design/src/Icon/Icon.tsx
@@ -19,7 +19,7 @@
 import React, { PropsWithChildren } from 'react';
 import styled from 'styled-components';
 
-import { space, color, borderRadius } from 'design/system';
+import { space, color, borderRadius, alignSelf } from 'design/system';
 
 export function Icon({
   size = 'medium',
@@ -61,6 +61,7 @@ const StyledIcon = styled.span`
   ${color};
   ${space};
   ${borderRadius};
+  ${alignSelf}
 `;
 
 export type IconSize = 'small' | 'medium' | 'large' | 'extraLarge' | number;
diff --git a/web/packages/shared/components/Notification/Notification.tsx b/web/packages/shared/components/Notification/Notification.tsx
index a227edbb40..b7cc7de607 100644
--- a/web/packages/shared/components/Notification/Notification.tsx
+++ b/web/packages/shared/components/Notification/Notification.tsx
@@ -157,7 +157,11 @@ const NotificationIcon = ({
   severity: NotificationSeverity;
   customIcon: React.ComponentType<IconProps>;
 } & IconProps) => {
-  const commonProps = { role: 'graphics-symbol', ...otherProps };
+  const commonProps = {
+    role: 'graphics-symbol',
+    alignSelf: 'baseline',
+    ...otherProps,
+  };
   if (CustomIcon) {
     return <CustomIcon {...commonProps} />;
   }
```
</details>

The only reason it wasn't already done this way is that when we originally implemented it in Connect, we couldn't get it to line up perfectly. It was mostly because of how the previous notification layout was structured and how our icons were made. The previous layout was something like this:

![previous layout](https://github.com/user-attachments/assets/515bcbe6-965c-4afc-9acc-13d2a24779d5)

We couldn't get the icon, which was in a separate flex than the title, to line up with the title itself. `align-items: flex-start` or `align-items: baseline` wasn't enough. Later on [we found a solution](https://github.com/gravitational/teleport/blob/069ebbcbaf6ba5bd372048c56d42699c591254f5/web/packages/teleterm/src/ui/Search/pickers/ResultList.tsx#L205-L206) (see [the story for Search Results](https://localhost:9002/?path=/story/teleterm-search--results)), but we've never got around to update how the notifications look like.

With your updates, it doesn't seem a problem anymore to get them to line up. I feel like they look somewhat better, especially the ones where the title spans multiple lines.

| Before | After |
| --- | --- |
| <img width="2560" alt="before" src="https://github.com/user-attachments/assets/85e1ea80-98df-4698-8110-f0c84f7d38db"> | <img width="2560" alt="after" src="https://github.com/user-attachments/assets/3f9a42d2-65af-46d3-970f-4faa2997ddf9"> |

Reply:
@ravicious This was done to remain consistent with the Figma designs that centers the icon in relation to the title+subtitle block. What I _could_ do, however, is to retain the centering, but limit the height of the icon container to the height of a single-line title plus a single-line subtitle. Let's see what it looks like.

Reply:
We discussed this over DMs. I wrote my comment without really taking the designs from Figma into account and left it based on how we currently use notifications, not how we want to use them in the future.

Most of the examples in the Storybook are a bit overblown. The one example that in my opinion looked the worst in the initial screenshots from Bartosz was the right Storybook column with notifications without a title. But this has also been fixed by Bartosz, since it turns out there was a Figma design for such notifications as well.